openstack/worker_pool.jinja

```diff
@@ -0,0 +1,29 @@
+{% from "virl.jinja" import virl with context %}
+
+{% set num_cpus = salt['grains.get']('num_cpus') %}
+{% set cluster_size = virl.cluster * sum([virl.compute1_active, virl.compute2_active, virl.compute3_active, virl.compute4_active]) %}
+{# picks machine size as an order of magniture of controller's cpu count #}
+{% set machine_size = sum([cluster_size > 0, num_cpus > 64, num_cpus > 32, num_cpus > 16, num_cpus > 8]) %}
+
+{% set worker_service = ['keystone', 'nova-api', 'nova-conductor', 'neutron', 'glance-api', 'glance-registry', 'cinder', 'mysql'] %}
+
+# Keystone has both admin and public workers, hence the numbers are halved for it
+# Mysql sets number of allowed connections, should be at most total * (max_size + overflow)
+
+{% set worker_counts = [
```

Comment:
These numbers are educated guesses on what ought to be enough. The machine_size determines what category to pick; I use the cluster size now just as a flag, but perhaps e.g. if a cluster has 4 computes, it should jump 2 sizes.